### PR TITLE
SCSS Coding Standards

### PR DESCRIPTION
**Pull request by Pushplaybang**
# Scope
CSS is a particularly difficult area to keep well organised as a project grows.  Standards help us  This proposal is based on our previous work toward an SCSS standard, which has been merged with the airBnB SCSS style guide.

## Whats Missing
We need a little more on the mobile first approach, use of a shame file, auto-prefixing and about using REMS.

I would also like to setup a scss linter yml file that will help us enforce this, as well as track code quality using static analysis over time.

@sharynt @JamesGarethSmith @darrynten @royscheeren @leogopal @RollYourOwnEd @Ricus-Rossouw @solnona  - fire at will.

### CODE REVIEW

File: SCSS-coding-standards.md [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+}
+```
+
+## CSS
+
+### Formatting
+
+* Use soft tabs (2 spaces) for indentation
+* Prefer dashes in class names.
+* Do not use ID selectors
```
Comment (solnona) [inline SCSS-coding-standards.md]: Maybe also try and stay away from HTML elements (h1, p, etc)

File: SCSS-coding-standards.md [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+  li {}
+  a {
+    &.special-one {}
+  }
+  
+}
+```
+
+These should rely on sensibly defaults, ideally set at a tag level.  By working from the most general styles, to the most specific, we able to lighten our CSS and leverage the cascade.  And by Modularising with parent classes, keep our html clean, our selectors light (low specificity) and these 'modules' re-usable.
+
```
Comment (solnona) [inline SCSS-coding-standards.md]: I don't agree 100%, there are a lot of cases where there needs to be a lot of overriding done to get rid of a style that is too general

File: SCSS-coding-standards.md (hunk 1 above)
Comment (Pushplaybang) [inline SCSS-coding-standards.md]: We've discussed offline, and I think we agree, my feeling is that we should only have sensible defaults declared on tag selectors, currently we tend to do a little too much here (for example in the default button styling)

Could add into the next section, not sure its appropriate for formatting specifically.

File: SCSS-coding-standards.md (hunk 2 above)
Comment (Pushplaybang) [inline SCSS-coding-standards.md]: Yeah, I think largely it's when we've failed to do this.

File: SCSS-coding-standards.md [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+      background: green;
+      font-weight: bold;
+      @include transition(background 0.5s ease);
+
+      .icon {
+        margin-right: 10px;
+      }
+    }
+    ```
+
```
Comment (Pushplaybang) [inline SCSS-coding-standards.md]: We need to add a section about media queries.

**Comment by Pushplaybang** [comment]
@Nona-Creative/nonabots  - please review the updated readme for scss and add comments if you feel it's necessary.  I think it contains alot of the basic stuff we've all discussed and aim for.

**Comment by Sntax** [comment]
@Nona-Creative/nonabots - I took the liberty of adding some minor additions and doing a little cleanup.

**Comment by Pushplaybang** [comment]
@Sntax  - sweet, though the TOC is autogenerated  so we'll need to correct the headings

**Comment by Sntax** [comment]
@Pushplaybang

> though the TOC is autogenerated

Derp!

**Comment by Pushplaybang** [comment]
all good - @Sntax @solnona @sharynt  - added notes about the shame file.  @Nona-Creative/nonabots can we merge?

I think this is looking good.   - Can we merge?

**Comment by Sntax** [comment]
Double thumbs - you're good to go.

**Pull request merged**